src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -827,25 +883,28 @@ private static RequiredHelperFunctions EmitTryMatchAtCurrentPosition(IndentedTex
 
             // In some limited cases, TryFindNextPossibleStartingPosition will only return true if it successfully matched the whole expression.
             // We can special case these to do essentially nothing in TryMatchAtCurrentPosition other than emit the capture.
-            switch (node.Kind)
-            {
-                case RegexNodeKind.Multi or RegexNodeKind.Notone or RegexNodeKind.One or RegexNodeKind.Set when !IsCaseInsensitive(node):
-                    // This is the case for single and multiple characters, though the whole thing is only guaranteed
-                    // to have been validated in TryFindNextPossibleStartingPosition when doing case-sensitive comparison.
-                    writer.WriteLine($"int start = base.runtextpos;");
-                    writer.WriteLine($"int end = start + {(node.Kind == RegexNodeKind.Multi ? node.Str!.Length : 1)};");
-                    writer.WriteLine("base.Capture(0, start, end);");
-                    writer.WriteLine("base.runtextpos = end;");
-                    writer.WriteLine("return true;");
-                    return requiredHelpers;
+            if ((node.Options & RegexOptions.RightToLeft) == 0)
```

Comment:
I'm not following why we can't do this same optimization when going right-to-left. If I have a pattern like `a` and use RightToLeft, shouldn't TryFindNextPossibleStartingPostion automatically give me the match by using a lastindexof so we should still optimize and just capture in the matching stage?

Reply:
We can, I just didn't.  I can.  I just need to fix up the math below.